zppy/utils.py

```diff
@@ -323,6 +327,17 @@ def check_parameter_defined(c: Dict[str, Any], relevant_parameter: str) -> None:
         raise ParameterNotProvidedError(relevant_parameter)
 
 
+def check_for_deprecated_parameters(
+    c: Dict[str, Any], deprecated_parameters: List[str]
+) -> None:
+    present_parameters = []
+    for parameter in deprecated_parameters:
+        if parameter in c.keys():
+            present_parameters.append(parameter)
+    if present_parameters:
+        raise DeprecatedParameterError(present_parameters)
```

Comment:
We could also just do a warning, but I figured an error is more likely to get the user's attention.

Reply:
does  `ts_fmt` a no-op or raise error before this change?

Reply:
on `main`, it just won't get used by anything. That is, if a user sets it thinking it's useful, they'll get no indication (either warning or error) that it's not actually useful. (Confirmed with cfg in https://github.com/E3SM-Project/zppy/pull/677).